### PR TITLE
Rc reporting

### PR DESCRIPTION
**Pull request by rajashree27**

### CODE REVIEW

File: src/components/Reporting/Reporting.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  return (
+    <div className={classes.root}>
+      <Grid container spacing={2}>
+        {QueryTypes.map((item) => (
+          <Grid key={item.id} item xs={6} sm={3}>
+            <Box className={classes.box}>
+              <Typography
+                variant="h6"
+                className={classes.name}
+                onClick={() => handleQueryClick(item)}
```
Comment (fszwedo) [inline src/components/Reporting/Reporting.tsx]: onclick on text is a bit misleading to the user - I was confused as I expected the whole 'button' container to be clickable

Let's add onclick to the box element instead

File: src/components/Reporting/ReportingDatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const [endDate, setEndDate] = useState<Date | null>(null);
+
+  const requiredStartDateError = startDate === null || startDate === undefined;
+  const requiredEndDateError =
+    endDate === null ||
+    endDate === undefined ||
+    startDate === null ||
+    startDate === undefined;
+  const helperText =
+    requiredEndDateError || requiredStartDateError ? "" : undefined;
```
Comment (fszwedo) [inline src/components/Reporting/ReportingDatePicker.tsx]: we should add a validation case for startDate > endDate

File: src/components/Reporting/ReportingDatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  const correctDateFormat = (date: any) => {
+    const [year, month, day] = date.split("-");
+    let updatedDate = `${year}-${month}-${day}`;
+    return updatedDate;
+  };
+
+  const handleStartDateChange = (event: any) => {
+    setStartDate(event.target.value);
+    localStorage.setItem("startDate", correctDateFormat(event.target.value));
```
Comment (fszwedo) [inline src/components/Reporting/ReportingDatePicker.tsx:30]: why do we need this in local storage? we don't restore the state after page refresh, I also don't think we should - so maybe we can get rid of that?

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+import axios from "axios";
+import Loader from "../Loader/Loader";
+import { QueryTypes } from "./QueryTypes";
+
+interface IPROPS {
+  selectedQuery: ReportingQuery;
+  closeHandler: Function;
+}
+
+const currentQuery = localStorage.getItem("queryType");
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: why do we take this from localstorage? shouldn't this be kept only till page refresh?

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+//Different URLs for different queries
+// const URL:string="";
+// if (currentQuery === "Purchase Events")
+//   URL = `http://localhost:3001/api/reporting/purchase-events`;
+// else
+//   URL = "";
+
+const URL = `http://localhost:3001/api/reporting/purchase-events`;
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: lets use backend url from constants.tsx here
I suggest storing the URL path in this file as well - so we have a central place to maintain backend routing dependencies

File: src/components/Reporting/ReportingDatePicker.tsx (hunk 2 above)
Comment (fszwedo) [reply src/components/Reporting/ReportingDatePicker.tsx]: and startDate cannot be in the future etc :D

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+      .post(URL, dataToSend, {
+        headers: {
+          "x-auth-token": `${localStorage.getItem("token")}`,
+        },
+      })
+      .then(function (response) {
+        setIsLoading(false);
+        setSubmitError(false);
+        removeLocalStorageItems();
+        console.log(response);
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: console log here and in error handling (line 94) I think are development process leftovers - lets remove this

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+          {isLoading && <Loader />}
+
+          {!isLoading && submitButtonClicked && !submitError && (
+            <Box
+              style={{
+                marginTop: "16px",
+                color: "#008000",
+                fontWeight: "bold",
+              }}
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: I'm no expert - but shouldn't we avoid inline styles? isn't it better to keep it centralized (I used module CSS, I see you were using other styling method but in this file the styling is applied inline in multiple places) and just apply appropriate classes on elements?

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+          )}
+
+          {!isLoading && submitButtonClicked && submitError && (
+            <Box
+              style={{
+                marginTop: "16px",
+                color: "red",
+              }}
+            >
+              Sorry there seems to be an error. Please try again after sometime!
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: lets rephrase to:
"There was an error in submitting your export request. Please try again or contact help@zoovu.com if the issue persists!"

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+                color: "#008000",
+                fontWeight: "bold",
+              }}
+            >
+              Thank you! Report will be sent to your email soon!
+            </Box>
+          )}
+
+          {!isLoading && submitButtonClicked && submitError && (
+            <Box
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: the box should not resize when content size changes - we should instead force the content to fit the container
what I refer to is that box width is low when we display accountid/date pickers, but is much wider when error/success message are displayed

File: src/components/Reporting/ReportingQueryModals.tsx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+          {!isLoading && submitButtonClicked && !submitError && (
+            <Box
+              style={{
+                marginTop: "16px",
+                color: "#008000",
+                fontWeight: "bold",
+              }}
+            >
+              Thank you! Report will be sent to your email soon!
```
Comment (fszwedo) [inline src/components/Reporting/ReportingQueryModals.tsx]: lets slightly rephrase to 
"Thank you! The requested report will be sent to your email soon!"

**Review by fszwedo — CHANGES_REQUESTED** [review]
@rajashree27 I left some comments for you - let me know in case of any questions

File: src/components/Reporting/ReportingQueryModals.tsx (hunk 8 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingQueryModals.tsx]: Done :)

File: src/components/Reporting/Reporting.tsx (hunk 1 above)
Comment (rajashree27) [reply src/components/Reporting/Reporting.tsx]: added onClick to the whole container as suggested

File: src/components/Reporting/ReportingQueryModals.tsx (hunk 5 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingQueryModals.tsx]: done

File: src/components/Reporting/ReportingDatePicker.tsx (hunk 3 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingDatePicker.tsx:30]: As discussed in our call, resolving this conversation as we are storing the startDate, endDate, accountId in the localStorage to send this data to the API endpoint

File: src/components/Reporting/ReportingDatePicker.tsx (hunk 2 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingDatePicker.tsx]: Added the validations :)

File: src/components/Reporting/ReportingQueryModals.tsx (hunk 4 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingQueryModals.tsx]: As discussed, I was using this to check which is the query type to hit that endpoint - but I removed this and will try to use another approach that you mentioned when we have more endpoints.

File: src/components/Reporting/ReportingQueryModals.tsx (hunk 9 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingQueryModals.tsx]: gave fixed size to container for the input and messages - loader box might change it's height a bit due to height of the loader

File: src/components/Reporting/ReportingQueryModals.tsx (hunk 7 above)
Comment (rajashree27) [reply src/components/Reporting/ReportingQueryModals.tsx]: I have removed the inline styles and also created separate files for CSS to remove all CSS from the components files

**Comment by rajashree27** [comment]
@fszwedo - I have completed all requested changes - you can check them and merge and please let me know if you have any questions

**Review by fszwedo — APPROVED** [review]

**Pull request merged**